content/en/docs/concepts/configuration/secret.md

```diff
@@ -127,15 +131,37 @@ data:
   password: MWYyZDFlMmU2N2Rm
 ```
 
-The data field is a map.  Its keys must consist of alphanumeric characters, '-', '_' or '.'.  The values are arbitrary data, encoded using base64.
-
-Create the secret using [`kubectl create`](/docs/reference/generated/kubectl/kubectl-commands#create):
+Now create the secret using [`kubectl create`](/docs/reference/generated/kubectl/kubectl-commands#create):
 
 ```shell
 $ kubectl create -f ./secret.yaml
 secret "mysecret" created
 ```
 
+For certain scenarios, you may wish to use the stringData field instead. An
+example of this is where the secret contains a configuration file with values
+populated via templating:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+stringData:
+  config.yaml: |-
+    apiUrl: "https://my.api.com/api/v1"
```

Comment:
Do you need the field, config.yaml?

Reply:
The reason I used an example like this was because at work we're using prometheus operator, and it uses a secret to define additional scrape targets (https://github.com/coreos/prometheus-operator/blob/master/example/additional-scrape-configs/additional-scrape-configs.yaml). The issue we had was that initially we didn't realise that `stringData` existed, so we had a two step process where we had to store a copy of the un-encoded file, and then encode it and update our secret yaml file.

Long story short, I was just trying to provide a semi-realistic example.

I could keep it consistent and just use the following instead:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
stringData:
  username: admin
  password: 1f2d1e2e67df
```

Do you think that would make more sense?

Reply:
This example is straightforward. The other example is okay too; I initially did not understand what information needed to be stored as a secret.